cfme/modeling/base.py

```diff
@@ -200,6 +200,21 @@ def exists(self):
         else:
             return True
 
+    def delete_if_exists(self, *args, **kwargs):
+        """Combines ``.exists`` and ``.delete()`` as a shortcut for ``request.addfinalizer``
+
+        Raises
+            NotImplementedError: If the ``.delete()`` is not implemented for the object
+        Returns: True if object existed and delete was initiated, False otherwise
+        """
+        if self.exists:
+            try:
+                self.delete(*args, **kwargs)
```

Comment:
OPTIONAL: I'd consider moving this to arg of delete call if it isn't sophisticated for sure. Something like `ignore_not_exists` or `ignore_errors`.

Reply:
@izapolsk Can you tell in a little more detail? I don't understand what you're suggesting.

Reply:
@valaparthvi I mean I lean towards considering such `delete_if_exists`  method as redundant. I'd better add option 'ignore_errors' to delete method.
it's worth discussing this. sure thing it depends on how many changes we will have to put in.

@mshriver ^^

Reply:
Ah, good point @izapolsk.  I'd like to be specific in this case, consider: `def delete(self, ignore_missing)` or `ignore_dne` or `ignore_exists`.

Reply:
Even more discussion was had offline - because of the common use of this method with `addfinalizer` calls, a separate method is useful in order to avoid lambda expressions.  Secondly, the implementation of `delete` itself is not the same across all entities, and we would not be defining it (or the kwarg) at a common place.